binheap.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+
+class BinHeap(object):
+
+    def __init__(self, iterable=None):
+        # heap will start at index 1 of heapList.
+        self.heapList = [0]
+        self._size = 0
+
+        # builds a heap from an iterable
+        if iterable is not None:
+            index = len(iterable) // 2
+            self._size = len(iterable)
+            self.heapList = self.heapList + list(iterable[:])
```

Comment:
This is a great idea. Good job! It can be dangerous if a generator is passed in, because this won't work, but the idea is solid!